exporters/trace/ocagent/src/main/java/io/opencensus/exporter/trace/ocagent/OcAgentTraceServiceStreamObservers.java

```diff
@@ -0,0 +1,245 @@
+            exportRequestObserver.onError(t); // Tries to notify OC-Agent about the error.
+            resetExportRequestObserverRef();
+          }
+        }
+
+        @Override
+        public void onCompleted() {
+          resetExportRequestObserverRef();
+        }
+      };
+
+  // Stream observer for UpdatedLibraryConfig messages. Once there's a new UpdatedLibraryConfig,
+  // apply it to the global TraceConfig.
+  private static final StreamObserver<UpdatedLibraryConfig> updatedLibraryConfigObserver =
+      new UpdatedLibraryConfigObserver(Tracing.getTraceConfig());
+
+  // A reference to the exportResponseObserver defined in this class.
+  // Could be shared between multiple threads.
+  static final AtomicReference<StreamObserver<ExportTraceServiceResponse>>
+      exportResponseObserverRef = new AtomicReference<>(exportResponseObserver);
+
+  // A reference to the updatedLibraryConfigObserver defined in this class.
+  // Could be shared between multiple threads.
+  static final AtomicReference<StreamObserver<UpdatedLibraryConfig>> updateConfigObserverRef =
+      new AtomicReference<>(updatedLibraryConfigObserver);
+
+  @VisibleForTesting
+  static class UpdatedLibraryConfigObserver implements StreamObserver<UpdatedLibraryConfig> {
+
+    // Inject a mock TraceConfig in unit tests.
+    private final TraceConfig traceConfig;
+
+    @VisibleForTesting
+    UpdatedLibraryConfigObserver(TraceConfig traceConfig) {
+      this.traceConfig = traceConfig;
+    }
+
+    @Override
+    public void onNext(UpdatedLibraryConfig value) {
+      TraceParams updatedTraceParams = TraceProtoUtils.getUpdatedTraceParams(value, traceConfig);
+      traceConfig.updateActiveTraceParams(updatedTraceParams);
+
+      /*@Nullable*/ StreamObserver<CurrentLibraryConfig> currentConfigObserver =
+          currentConfigObserverRef.get();
+      // Add a check in case that connection dropped while config is being updated.
+      if (currentConfigObserver != null) {
+        // Bouncing back CurrentLibraryConfig to Agent.
+        io.opencensus.proto.trace.v1.TraceConfig currentTraceConfigProto =
+            TraceProtoUtils.getCurrentTraceConfig(traceConfig);
+        CurrentLibraryConfig currentLibraryConfig =
+            CurrentLibraryConfig.newBuilder().setConfig(currentTraceConfigProto).build();
+        currentConfigObserver.onNext(currentLibraryConfig);
+      }
+    }
+
+    @Override
+    public void onError(Throwable t) {
+      // If there's an error, reset the reference to currentConfigObserver to force
+      // re-establish a connection.
+      // TODO(songya): add retries on this connection before starting a new connection.
+      logger.log(Level.WARNING, "Config stream is disconnected.", t);
+      /*@Nullable*/ StreamObserver<CurrentLibraryConfig> currentConfigObserver =
+          currentConfigObserverRef.get();
+      if (currentConfigObserver != null) {
+        currentConfigObserver.onError(t); // Tries to notify OC-Agent about the error.
```

Comment:
Why this?

Reply:
This is for cases when Exporter saw errors in the stream but Agent hadn't seen, and Exporter would try to notify Agent by calling `onError`. Anyway this is not needed since when there're errors in the stream, stream is likely to be interrupted and `onError` won't work.